File: testes/projetojava/Atleta.java
import java.util.Scanner;

public class Atleta {
    private static final int INFORMATICA = 1;
    private static final int ELETROELETRONICA = 2;

    private static final int ANO_ESCOLAR_1 = 1;
    private static final int ANO_ESCOLAR_2 = 2;
    private static final int ANO_ESCOLAR_3 = 3;

    private Scanner scanner = new Scanner(System.in);

    private String nome;
    private int curso;
    private String nomeCurso;
    private int ano;
    private String dataNascimento;

    public Atleta() {
        limparConsole();
        System.out.println("Digite o nome do atleta:");
        nome = scanner.next();
        System.out.println("Digite o curso do atleta:");
        System.out.println("1 - Informática");
        System.out.println("2 - Eletroeletrônica");
        curso = lerOpcao(1, 2);
        nomeCurso = curso == INFORMATICA ? "Informática" : "Eletroeletrônica";
        System.out.println("Digite o ano escolar do atleta (1, 2 ou 3):");
        ano = lerOpcao(1, 3);
        System.out.println("Digite a data de nascimento do atleta (DD/MM/AAAA):");
        dataNascimento = scanner.next();
        // Aqui você poderia validar se a data de nascimento está em um formato válido.
    }

    private void limparConsole() {
        System.out.print("\033[H\033[2J");
        System.out.flush();
    }

    private int lerOpcao(int min, int max) {
        int opcao;
        do {
            opcao = scanner.nextInt();
            if (opcao < min || opcao > max) {
                System.out.println("Opção inválida, digite novamente!");
            }
        } while (opcao < min || opcao > max);
        return opcao;
    }

    public String getNome() {
        return nome;
    }

    public String getNomeCurso() {
        return nomeCurso;
    }

    public int getAno() {
        return ano;
    }

    public String getDataNascimento() {
        return dataNascimento;
    }
}

File: testes/projetoc++/Atleta.cpp
#include "Atleta.h"
#include <iostream>

Atleta::Atleta() {
    limparConsole();
    std::cout << "Digite o nome do atleta: ";
    std::cin >> nome;
    std::cout << "Digite o curso do atleta:" << std::endl;
    std::cout << "1 - Informática" << std::endl;
    std::cout << "2 - Eletroeletrônica" << std::endl;
    curso = lerOpcao(1, 2);
    nomeCurso = (curso == INFORMATICA) ? "Informática" : "Eletroeletrônica";
    std::cout << "Digite o ano escolar do atleta (1, 2 ou 3): ";
    ano = lerOpcao(1, 3);
    std::cout << "Digite a data de nascimento do atleta (DD/MM/AAAA): ";
    std::cin >> dataNascimento;
    // Aqui você poderia validar se a data de nascimento está em um formato válido.
}

std::string Atleta::getNome() {
    return nome;
}

std::string Atleta::getNomeCurso() {
    return nomeCurso;
}

int Atleta::getAno() {
    return ano;
}

std::string Atleta::getDataNascimento() {
    return dataNascimento;
}

void Atleta::limparConsole() {
    // Esta função não tem um equivalente direto em C++, já que a limpeza do console é mais complexa em sistemas UNIX.
    // Você pode ignorar esta função, se desejar.
}

int Atleta::lerOpcao(int min, int max) {
    int opcao;
    do {
        std::cin >> opcao;
        if (opcao < min || opcao > max) {
            std::cout << "Opção inválida, digite novamente!" << std::endl;
        }
    } while (opcao < min || opcao > max);
    return opcao;
}
